pkg/validate/rsync/validate/source_spec_validator.go

```diff
@@ -301,6 +303,57 @@ func OverrideSpec(override *v1beta1.OverrideSpec, syncKind string) status.Error
 	return nil
 }
 
+// ReconcilerName validates the reconciler name.
+func ReconcilerName(reconcilerName string) status.Error {
+	if errs := validation.IsDNS1123Subdomain(reconcilerName); errs != nil {
+		return InvalidReconcilerName(reconcilerName, strings.Join(errs, ", "))
+	}
+	return nil
+}
+
+// RootSyncName validates the RootSync name length.
+func RootSyncName(rs *v1beta1.RootSync) status.Error {
+	nameLength := len(rs.Name)
+	if nameLength > reconcilermanager.MaxRootSyncNameLength {
+		return RootSyncNameLengthExceeded(nameLength)
+	}
+	return nil
+}
+
+// RootSyncMetadata validates name, namespace, and annotations.
+func RootSyncMetadata(rs *v1beta1.RootSync) status.Error {
```

Comment:
Please add unit tests for `RootSyncMetadata` and `RepoSyncMetadata`

Reply:
Done